youtube_dl/extractor/youtube.py

```diff
@@ -1553,25 +1557,37 @@ def _extract_playlist(self, playlist_id):
                     'Invalid parameters. Maybe URL is incorrect.',
                     expected=True)
             elif re.match(r'[^<]*Choose your language[^<]*', match):
+                # Looks good; continue
                 continue
             else:
                 self.report_warning('Youtube gives an alert message: ' + match)
 
-        # Extract the video ids from the playlist pages
         def _entries():
+            # Extract the video ids from the playlist pages
             more_widget_html = content_html = page
+
             for page_num in itertools.count(1):
+                # Loop to find videos until break
+
                 matches = re.finditer(self._VIDEO_RE, content_html)
-                # We remove the duplicates and the link with index 0
-                # (it's not the first video of the playlist)
-                new_ids = orderedSet(m.group('id') for m in matches if m.group('index') != '0')
-                for vid_id in new_ids:
-                    yield self.url_result(vid_id, 'Youtube', video_id=vid_id)
 
+                # Make list of videos
+                new_videos = [{'id': m.group('id'),
+                               'title': m.group('title').strip()}
+                              for m in matches
+                              if m.group('index') != '0'  # Ignore link with index 0
+                              if m.group('title').strip()]  # Ignore links without titles, which also prevents duplicates
```

Comment:
Even if all videos have a title the regex may stop to detect the title someday and I prefer getting all the videos in the playlist instead of missing some or all of them.

Also, I haven't tested but titles could be empty or have only spaces (I don't know if YouTube checks that before allowing you to submit the video).